src/menu/menu.js

```diff
@@ -27,20 +31,51 @@ export function getRestaurantMenu() {
         if (!Array.isArray(cart)) {
           cart = [];
         }
-        cart.push(item);
+
+        //get id from URL
+        const id = window.location.hash.split("/")[1];
+        console.log("the id is ", id);
+
+        const menuItem = { item, id };
+        cart.push(menuItem);
         localStorage.setItem("cart", JSON.stringify(cart));
 
+        totalQuantity++;
+        totalPrice += item.price;
+
         alert(`${item.name} has been added to your cart!`);
```

Comment:
Since we have this alert and show the user that the item was added to added we should add a check to make sure no error was thrown. 
So this alert should be showed only after making sure the localStorage.setItem didn't throw any errors.
```
try {
  localStorage.setItem("name", "Hello World!");
} catch (e) {
    alert(`Local storage error: ${e.message}`);
}
```

https://chrisberkhout.com/blog/localstorage-errors/